api/api_public_blockchain.go

```diff
@@ -63,13 +63,23 @@ func (s *PublicBlockChainAPI) BlockNumber() *big.Int {
 }
 
 // ChainID returns the chain ID of the chain from genesis file.
+// This will be deprecated.
 func (s *PublicBlockChainAPI) ChainID() *big.Int {
 	if s.b.ChainConfig() != nil {
 		return s.b.ChainConfig().ChainID
 	}
 	return nil
 }
 
+// ChainId returns the chain ID of the chain from genesis file.
+// This is for compatibility with ethereum client
+func (s *PublicBlockChainAPI) ChainId() *big.Int {
```

Comment:
Why don't you call call `s.ChainID()` internally?

Reply:
Why do we call it? ㅋㅋㅋㅋ

Reply:
Top 4 reasons why code duplication is harmful:
1. Duplicate code makes your program lengthy and bulky :
2. Duplication decreases your code quality :
3. Shotgun Surgery:
4. Increases security risks:
from https://www.codegrip.tech/productivity/what-is-duplicate-code/

Reply:
Thanks. I removed duplicated code. :)